parsers/AX.py

```diff
@@ -0,0 +1,388 @@
+    # Åland administration does not allow export
+    # to Finland through this cable
+    FIFlow = []
+    for x in range(760, 815-6):
+        for abr in shorts:
+            im1 = im.crop((x, 328, x+6, 337))
+            if im1 == mapping[abr]:
+                FIFlow.append(abr)
+    FIFlow = "".join(FIFlow)
+    FIFlow = round(float(FIFlow),1)
+
+
+
+    # Consumption
+    Cons = []
+    for x in range(650, 700-6):
+        for abr in shorts:
+            im1 = im.crop((x, 564, x+6, 573))
+            if im1 == mapping[abr]:
+                Cons.append(abr)
+    Cons = "".join(Cons)
+    Cons = round(float(Cons),1)
+
+    # Wind production
+    WProd = []
+    for x in range(650, 700-6):
+        for abr in shorts:
+            im1 = im.crop((x, 576, x+6, 585))
+            if im1 == mapping[abr]:
+                WProd.append(abr)
+    WProd = "".join(WProd)
+    WProd = round(float(WProd),1)
+
+    # Fossil fuel production
+    FProd = []
+    for x in range(650, 700-6):
+        for abr in shorts:
+            im1 = im.crop((x, 588, x+6, 597))
+            if im1 == mapping[abr]:
+                FProd.append(abr)
+    FProd = "".join(FProd)
+    FProd = round(float(FProd),1)
+    
+    # Flows and consumption do not match
+    # unless the flow to Gustafs is export to Finland
+    FIFlow = FIFlow-GustafsFlow
+    
+    # Calculate sum of exchanges
+    SumExchanges = SE3Flow+FIFlow
+    
+    # Calculate total production
+    TotProd = Cons-SumExchanges
+    
+    # The production that is not fossil fuel based can be assumed to be bio
+    BProd = TotProd - WProd - FProd
+    
+    obj = dict({'production':TotProd,'consumption':Cons,'wind':WProd,
+               'fossil':FProd,'biomass':BProd,'SE3->AX':SE3Flow,
+               'FI->AX':FIFlow,'fetchtime':fetchtime})
+    
+    return obj
+
+
+def fetch_production(country_code='AX', session=None):
+    """Requests the last known production mix (in MW) of a given country
+
+    Arguments:
+    country_code (optional) -- used in case a parser is able to fetch multiple countries
+    session (optional)      -- request session passed in order to re-use an existing session
+
+    Return:
+    A dictionary in the form:
+    {
+      'countryCode': 'FR',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
+      },
+      'storage': {
+          'hydro': -10.0,
+      },
+      'source': 'mysource.com'
+    }
+    """
+    obj = _fetch_data(session)
+
+    data = {
+        'countryCode': country_code,
+        'production': {},
+        'storage': {},
+        'source': 'kraftnat.aland.fi',
+        'datetime': arrow.get(obj['fetchtime']).datetime
+    }
+    data['production']['biomass'] = round(obj['biomass'],1)
```

Comment:
We try to avoid assuming fuel types in general. Better to set something to 'unknown' if we can't be sure.